src/scipp/experimental/plotting/wrappers.py

```diff
@@ -24,6 +24,24 @@ def _to_data_array(obj):
     return out
 
 
+def _convert_if_not_none(x, unit):
+    if x is not None:
+        return to_unit(x, unit=unit)
+    return x
+
+
+def _preprocess(obj, crop):
+    out = _to_data_array(obj)
+    crop = {} if crop is None else crop
+    for dim, sl in crop.items():
+        smin = _convert_if_not_none(sl.get('min'), unit=out.meta[dim].unit)
+        smax = _convert_if_not_none(sl.get('max'), unit=out.meta[dim].unit)
+        start = max(out[dim, :smin].sizes[dim] - 1, 0)
+        width = out[dim, smin:smax].sizes[dim]
+        out = out[dim, start:start + width + 2]
```

Comment:
Translating label indices into integer indices? And the +2 is for inclusion of the extra point?

Reply:
Yes, but I don't know how efficient that is. What do you think?

Reply:
Why extra at lower bound?

It should be quite efficient. Only overhead is slicing twice, but unless you have huge numbers of coords/masks/attrs that should not be relevant.